### PR TITLE
Task 3: State management, Redux

### PR DESCRIPTION
**Pull request by jkyrychenko**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -15,26 +18,18 @@ import Registration from './components/Registration/Registration';
...
 import isTokenExist from './mixins/token';
 
 const App = () => {
+	const dispatch = useDispatch();
 	const [isLoggedIn, setIsLoggedIn] = useState(isTokenExist);
-	const [userName, setUserName] = useState('');
-	const { loading, data } = useFetch('http://localhost:3000/courses/all');
-	const authors = useFetch('http://localhost:3000/authors/all').data;
+	const fetchedCourses = useFetch('http://localhost:3000/courses/all');
+	const fetchedAuthors = useFetch('http://localhost:3000/authors/all');
```
Comment (marianasvirzhevska) [inline src/App.jsx]: Consider this
```suggestion
	const { courses, loading: coursesLoading } = useFetch('http://localhost:3000/courses/all');
	const { authors, loading: authorsLoading } = useFetch('http://localhost:3000/authors/all');
```

File: src/App.jsx [hunk 2]
```diff
@@ -44,25 +39,35 @@ const App = () => {
...
+						token: localStorage.getItem('userToken'),
+					})
+				);
+			});
 		}
-	}, [isLoggedIn]);
+
+		dispatch(setAuthors(fetchedAuthors.data));
+		dispatch(setCourses(fetchedCourses.data));
+	}, [isLoggedIn, fetchedAuthors, fetchedCourses]);
```
Comment (marianasvirzhevska) [inline src/App.jsx]: And this
```suggestion
		dispatch(setAuthors(authors));
		dispatch(setCourses(courses));
	}, [dispatch, isLoggedIn, authors, courses]);
```

File: src/App.jsx [hunk 3]
```diff
@@ -44,25 +39,35 @@ const App = () => {
...
+	useEffect(() => {
+		if (fetchedCourses.loading || fetchedAuthors.loading) {
+			setIsLoading(true);
+		}
+
+		let loadingTimer = setTimeout(() => setIsLoading(false), 1000);
+		return () => {
+			clearTimeout(loadingTimer);
+		};
+	}, [isLoading]);
```
Comment (marianasvirzhevska) [inline src/App.jsx]: And this
Avoid timers
```suggestion
	useEffect(() => {
		setIsLoading(coursesLoaded && authorsLoaded);
	}, [coursesLoaded, authorsLoaded]);
```

File: src/store/authors/reducer.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { actionTypes } from './actionTypes';
+
+const authorsInitialState = {
+	authors: [],
+};
+
+const authorsReduser = (state = authorsInitialState, { type, payload }) => {
```
Comment (marianasvirzhevska) [inline src/store/authors/reducer.js]:
```suggestion
const authors = (state = authorsInitialState, { type, payload }) => {
```

File: src/store/index.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { combineReducers } from 'redux';
+import authorsReduser from './authors/reducer';
```
Comment (marianasvirzhevska) [inline src/store/index.js]: and 
```suggestion
import authors from './authors/reducer';
```

File: src/store/index.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+import coursesReduser from './courses/reducer';
+import userReduser from './user/reducer';
+
+const reducers = combineReducers({
+	allAuthors: authorsReduser,
+	allCourses: coursesReduser,
+	user: userReduser,
```
Comment (marianasvirzhevska) [inline src/store/index.js]:
```suggestion
	authors,
	courses,
	user,
```

File: src/App.jsx (hunk 1 above)
Comment (jkyrychenko) [reply src/App.jsx]: fixed

File: src/App.jsx (hunk 2 above)
Comment (jkyrychenko) [reply src/App.jsx]: fixed

File: src/App.jsx (hunk 3 above)
Comment (jkyrychenko) [reply src/App.jsx]: fixed

File: src/store/authors/reducer.js (hunk 4 above)
Comment (jkyrychenko) [reply src/store/authors/reducer.js]: fixed

File: src/store/index.js (hunk 5 above)
Comment (jkyrychenko) [reply src/store/index.js]: fixed

File: src/store/index.js (hunk 6 above)
Comment (jkyrychenko) [reply src/store/index.js]: fixed

File: src/App.jsx (hunk 3 above)
Comment (jkyrychenko) [reply src/App.jsx]: fixed

File: src/store/index.js (hunk 5 above)
Comment (jkyrychenko) [reply src/store/index.js]: fixed

File: src/store/index.js (hunk 6 above)
Comment (jkyrychenko) [reply src/store/index.js]: fixed

**Pull request merged**